Me: Use SitesDropdown instead of SelectSite in /me/account

ebinnion opened the pull request:
Fixes #451

Now that we have a `SitesDropdown`, we can retire our use of `SelectSite`. This component swaps out the former for the latter on the `/me/account` route.

__Note:__ There is still some jankiness where switching a site and then refreshing the page may not show the proper primary site. This is because the sites list data is stale after we make the change. So, we'll need to find some way to update the primary site in the sites list.

To test:
- Checkout `fix/me-account-site-dropdown`
- Go to `/me/account`
- Change primary site
- Go to `wordpress.com/me/account`
- Does primary site reflect the change that you just made?

![screen shot 2](https://cloud.githubusercontent.com/assets/1126811/11905272/23a10104-a58b-11e5-8d4d-219f7d48c211.png)

ockham commented:
Now that #1886 has been merged, this is the last occurrence of `me/select-site.jsx`, so maybe you could also remove that file as part of this PR?

mtias commented:
Nice.

> This is because the sites list data is stale after we make the change. So, we'll need to find some way to update the primary site in the sites list.

This is something we'll properly fix once we migrate site-settings to redux and centralize the data. In the meantime you could just set the primary attribute manually, or change the order of the array. <a href="https://github.com/Automattic/wp-calypso/blob/master/client/lib/sites-list/list.js#L175">This is where we set the attribute</a>.

cc @rralian to keep in mind for settings.

ebinnion commented:
@mtias - I updated the PR to manually set the primary site. After I did that, I noticed that the old primary site still showed on refresh, which I tracked down to `SitesDropdown` falling back to the primary site if a site is not specified. So, the primary site was being shown until the `/me/settings` API call came back.

I addressed by adding an `isPlaceholder` prop to `SitesDropdown` which, when `true`, will display a `Site` placeholder.

After that, I noticed that the site and chevron could overlap each other. So, I adjusted the styling of the component to use flexbox.

*Overlapping:*
![screen shot 5](https://cloud.githubusercontent.com/assets/1126811/12339131/eb4fe0f0-bad8-11e5-9188-8bf3232ecced.png)

*Overlapping fixed:*

Chrome:
![screen shot 8](https://cloud.githubusercontent.com/assets/1126811/12339183/3e4a2004-bad9-11e5-9464-e1cd2f1943bb.png)

![screen shot 6](https://cloud.githubusercontent.com/assets/1126811/12339188/49af33b2-bad9-11e5-84d9-308f6f701798.png)


IE11:

![screen shot 9](https://cloud.githubusercontent.com/assets/1126811/12339192/4e643998-bad9-11e5-846b-2386bfc95123.png)

![screen shot 7](https://cloud.githubusercontent.com/assets/1126811/12339194/5113a35e-bad9-11e5-88a9-ae5fa2854a9a.png)


iPhone4

![screen shot 4](https://cloud.githubusercontent.com/assets/1126811/12339067/a6528fd4-bad8-11e5-9b8f-b606348b727a.png)
 Safari:



@ockham - I went ahead and removed the `select-site` component in this PR as well. :+1:

rickybanister commented:
Looks good to me.

dmsnell commented on client/me/account/index.jsx:
```diff
@@ -210,6 +210,13 @@ module.exports = React.createClass( {
 
+	onSiteSelect( siteSlug ) {
+		let selectedSite = sites.getSite( siteSlug );
+		if ( sites.getSite( siteSlug ) ) {
```
is this a typo? should it be:

```js
if ( selectedSite ) {
```

ebinnion replied:
heh... you're right :smile: I created the variable so I didn't have to get the site twice, and then I did it anyways :panda_face:

dmsnell commented on client/me/account/index.jsx:
```diff
@@ -210,6 +210,13 @@ module.exports = React.createClass( {
+	onSiteSelect( siteSlug ) {
+		let selectedSite = sites.getSite( siteSlug );
+		if ( sites.getSite( siteSlug ) ) {
+			this.props.userSettings.updateSetting( 'primary_site_ID', selectedSite.ID );
```
Do we not have a Flux-y action to update the primary site? This looks dangerously entangled here.

ebinnion replied:
This is specifically for the `userSettings` module which doesn't have flux. The way we handle data for `user` and `userSettings` could definitely use some work.

dmsnell commented:
@ebinnion looks good. My only real question was about a possible typo and the data path for updating the primary site.
